### ISSUE TITLE
Error Reports from Contact-Us improvements

### ISSUE THREAD

**Issue by ACowan0105** [virtualcell/vcell#835]
Would like:

1. enable users to add to message
2. want to identify user so can give feedback
3. improve logging and include database transactions

**Comment by jcschaff** [comment]
Partly related: #1495 covers the *legibility* of these emails and is addressed by #2014 (the body is now rendered as text, and a version 2 report keeps the log, model and recorded events in separate fields instead of stuffing them into an exception message).

Item 1 (*enable users to add to message*) and item 2 (*identify user so can give feedback*) are **not** addressed there. Both fields already exist on the report and are rendered in the email — `message` and `username` — but auto-submitted crash reports leave them empty, so what is missing is the client-side prompt, not the transport.

**Comment by jcschaff** [comment]
Reopening — closed automatically by the merge of #2014, which had "fix #835 items 1-2" in its title; GitHub read the closing keyword and ignored the qualifier. My mistake.

**Done in #2014:**
1. *Enable users to add to message* — the error dialog now has an optional "What were you doing? (optional)" field, sent as the report's `message`.
2. *Identify user so can give feedback* — reports were arriving anonymous because `ErrorUtils` was told who was logged in only once at startup, from the command-line user name, which is normally absent since people log in through the dialog. It is now refreshed on each successful connect, so the report carries the account.

**Still open:**
3. *Improve logging and include database transactions* — not addressed. The report now has a clean place to put it: a version 2 report carries `modelInfo`, `clientLog` and `userEvents` as separate fields, so a transactions section would be another field rather than more text concatenated into the log.

**Comment by jcschaff** [comment]
Closing. Items 1 and 2 are done, and item 3 belongs to a different system than it did when this was filed in 2023.

**Done in #2014:**
1. *Enable users to add to message* — the error dialog now carries an optional "What were you doing? (optional)" field, sent as the report's `message`. A log records what VCell did; only the user can say what they were trying to do.
2. *Identify user so can give feedback* — reports had been arriving anonymous. `ErrorUtils` was told who was logged in only once, at startup, from the command-line user name, which is normally absent because people log in through the dialog. It is now refreshed on each successful connect, so a report carries the account.

**Item 3 — *improve logging and include database transactions* — is out of scope here, deliberately.**

Contact-Us reports are *client* reports: what one user's desktop client was doing when it failed. Server error logs do not go to `vcell_support`, and should not — that mailbox is for users to reach us, not a destination for server telemetry.

Server-side logging and monitoring have also been rebuilt since this was filed, and are no longer something an error email would improve:

- **Structured logging.** `vcell-rest` emits Elastic Common Schema JSON (#1312), so logs aggregate and query as fields rather than as text.
- **Noise removed.** Several sources of high-volume log spam were traced and fixed rather than tolerated — most sharply the export-progress `Infinity` fault that produced roughly 4,900 error lines a minute until the pods were rolled. Access tokens are also no longer written to the logs.
- **Loki in the cluster.** All services ship to Loki, queryable across namespaces instead of by shelling into a pod.
- **Direct access.** `logcli` from the command line, a Claude skill wrapping it for log investigations, and the hosted Grafana UI for browsing and dashboards.

That covers what item 3 was reaching for — the ability to see what the server was doing around a failure — through infrastructure built for the purpose, rather than by attaching more content to a support email.

If a specific need remains for database-transaction detail in a *client* report, that is worth its own issue: the version 2 report added in #2014 gives it a clean home, carrying `modelInfo`, `clientLog` and `userEvents` as separate fields, so a transactions section would be another field rather than more text concatenated into the log.
